### ISSUE TITLE
[MNG-193] symmetry for outputs of a plugin

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#2784]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** opened **[MNG-193](https://issues.apache.org/jira/browse/MNG-193?redirect=false)** and commented

We need a way to declaratively say what the outputs of a plugins are going to be. The concrete example being the Antlr plugin that generates a source root, we should know what this is going to be before executing the plugin.

---

**Issue Links:**
- [MNG-1569](https://issues.apache.org/jira/browse/MNG-1569) Make build process info read-only to mojos, and provide mechanism for explicit out-params for mojos to declare

- [MNG-2333](https://issues.apache.org/jira/browse/MNG-2333) Plugins need to offer up all information without executing
 (_**"is depended upon by"**_)
- [MNG-125](https://issues.apache.org/jira/browse/MNG-125) guarded mojo execution
 (_**"is depended upon by"**_)

**Remote Links:**
- [Page
](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=65875544)

1 votes, 2 watchers

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

is this still something we're working toward for 2.0, or is it something for later?

**Comment by jira-importer** [comment]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** commented

The use cases I can think of:

o generated sources being added the compile source roots

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

waiting on a proposal, more use cases.

**Comment by jira-importer** [comment]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** commented

This will be part of a new Plugin API which is out of scope for Maven 3.0.

**Comment by jira-importer** [comment]
**[Elliotte Rusty Harold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=elharo)** commented

We've lived this long without it. I'm guessing it's not needed. Anyone disagree?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Agree with [Elliotte Rusty Harold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=elharo), what real world problem are we trying to solve?
